pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -653,18 +653,41 @@ fi
 # If we're not just building, then make the new configuration the boot
 # default and/or activate it now.
 if [[ "$action" = switch || "$action" = boot || "$action" = test || "$action" = dry-activate ]]; then
+    # Using systemd-run here to protect against PTY failures/network
+    # disconnections during rebuild.
+    # See: https://github.com/NixOS/nixpkgs/issues/39118
+    cmd=(
+        "systemd-run"
+        "-E" "LOCALE_ARCHIVE"
+        "-E" "NIX_PATH"
```

Comment:
`switch-to-configuration` should not use `NIX_PATH` because everything has already been built. At this point nothing should need to call Nix commands anymore.

```suggestion
```

Reply:
Good point. I actually extract those values from my `nixos-rebuild` wrapper (from #39118), and it needs there because of the build phase. But in this case not.